app/uk/gov/hmrc/nisp/services/StatePensionService.scala

```diff
@@ -77,4 +84,13 @@ class StatePensionService @Inject()(statePensionConnector: StatePensionConnector
       throw new RuntimeException(s"Un-accounted for exclusion in NispConnectionNI: $exclusions")
     }
   }
+
+  private def getDateWithRegex(copeResponse: String): LocalDate = {
+    val copeResponseDateCapturingRegex: Regex = """(?:.*)(?:'\{"errorCode":"EXCLUSION_COPE_PROCESSING","copeDataAvailableDate":\")(\d{4}-\d{2}-\d{2})(?:\"}')""".r
```

Comment:
This is a little brittle, I would consider having a case class that represents the repsonse and parse it that way then extract the date

Reply:
Ticket created to refactor this.